### ISSUE TITLE
'test -w' contradicts actual mutability of files on FAT32 partitions

### ISSUE THREAD

**Issue by cochon-git** [rmyorston/busybox-w32#241]
Taking note of the limitations section in README.md I'm not sure if this should be a bug, a change request or a 'no can do', but it would be useful if the 'fake' owner returned for a file on a file system that has no ownership concept (FATnn) could be the current user and not 'root', so that the fake uid/gid/mask matches the actual mutability of the file. By extension 'root' IF running elevated UAC.

Currently both 'test -w' and parsing the output of 'ls -l' indicate that files on FAT32 are not writable to the local [creating] user, when in fact they are.

There may of course be a good reason for the way it is, as it skirts the *nix 'mount -o uid=1234' concept which doesn't apply on Windows. At least on *nix it is consistent and file writes will also fail if UID/GID mismatches. 

See example below run from a non-Administrator local account.

```
E:\Scratch>busybox-w32-PRE-4616-ge6238530e.exe ash -l
Y:/home/philg $ cat ~/tmp/test.sh
#!/bin/sh

uname -a
df -T I: J: # I is NTFS J is FAT32

echo 'testing' > I:/test.txt || echo Ooops
echo 'testing' > J:/test.txt || echo Ooops

ls -al I:/test.txt J:/test.txt

for file in I:/test.txt J:/test.txt; do
  [ -w "$file" ] \
    && echo "$file is writeable" \
    || echo "$file is NOT writeable"
  echo 'more' >> "$file" && echo "Wrote more to $file"
done

ls -l I:/test.txt J:/test.txt
```

```
Y:/home/philg $ ~/tmp/test.sh
Windows_NT piglet 10.0 19043 i686 MS/Windows
Filesystem           Type       1K-blocks      Used Available Use% Mounted on
I:                   NTFS         7292484     27044   7265440   0% I:/
J:                   FAT32         536576        20    536556   0% J:/
-rw-rw-r--    1 philg    philg            8 Feb 16 18:42 I:/test.txt
-rw-rw-r--    1 root     root             8 Feb 16 18:42 J:/test.txt
I:/test.txt is writeable
Wrote more to I:/test.txt
J:/test.txt is NOT writeable
Wrote more to J:/test.txt
-rw-rw-r--    1 philg    philg           13 Feb 16 18:42 I:/test.txt
-rw-rw-r--    1 root     root            13 Feb 16 18:42 J:/test.txt

```

**Comment by rmyorston** [comment]
I've determined that filesystems which don't support ownership data return a null [SID](https://docs.microsoft.com/en-gb/windows/win32/secauthz/well-known-sids).

I've used that as an indication busybox-w32 should treat such files as belonging to the current user. This applies to FAT filesystems and things like CD-ROMs. Obviously you still can't write to CD-ROMs but that's handled by permissions, not ownership.

When running with elevated privileges files still seem to belong to the current user, not root.

There's a new [prerelease](https://frippery.org/files/busybox/prerelease/) for anyone who'd like to try it.
